dnf/cli/output.py

```diff
@@ -2250,3 +2261,119 @@ def progressbar(current, total, name=None):
         sys.stdout.write('\n')
 
     sys.stdout.flush()
+
+class PackageFormatter(object):
+    """
+    # :api
+    Wrapper for dnf.package.Package, so we can control formatting
+    in --queryformat
+    """
+    def __init__(self, pkg):
+        self._pkg = pkg
+
+    def __format__(self, fmt):
+        return self.format(fmt)
+
+    def _attr(self, name):
+        name = name.lower()
+        if name not in QUERY_TAGS:
+            raise AttributeError("Unknown tag '%s'" % name)
```

Comment:
I think it should rather raise ```KeyError``` like ```str.format```. ```AttributeError``` implies that there is no attribute called `_attr`.